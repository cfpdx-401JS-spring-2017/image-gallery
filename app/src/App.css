.App {
  text-align: center;
  display: flex;
  height: 100vh;
  width: 100%;
  justify-content: center;
  align-items: center;

}

.ThumbNail {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  height: 30vh;
}

.List {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

button {
  height: 10vh;
  width: 20vw;
  background-color: #476A6F;
  border: none;
  margin: 2vh 2vh;
  box-shadow: 1px 1px 1px 1px #202A25;
  border-radius: 10px;
}

.Link {
  color: #F7F0F0;
  text-decoration: none;
  font-size: 2em;
}

.Link:hover {
  
}
