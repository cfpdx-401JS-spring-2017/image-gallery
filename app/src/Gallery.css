
.YoungDivvy {
    height: 100%;
    width: 100vw;
    background-color: #84A98C;
    font-size: 3em;
    position: absolute;
    top: 0;
    left:0;
    z-index: 2;
    display: flex;
    justify-content: center;
    align-items: center
}

.mainButtons {
    text-decoration: none;
    background-color: #84A98C;
    color: #2F3E46;
    border: 5px solid #2F3E46;
    border-radius: 15px;
    margin: 2vh 2vh;
    width: 50vw;
    height: 20vh;
    font-size: 3em;
}

img {
    width: 50vw;
    max-height: 60vh;
    border: 5px solid white;
    border-radius: 10px;
}

.arrows {
    border-radius: 100%;
    border: 5px solid white;
    width: 10vh;
    height: 10vh;
    padding-top: 2vh;
    margin: 3vh;
    z-index: 4;
 
}

h5 {
    font-size: .5em;
}

.ImageTitle {
    font-size: .7em;
}

.exitButton  {
    position: absolute;
    top: 0vh;
    left: 90vw;
}