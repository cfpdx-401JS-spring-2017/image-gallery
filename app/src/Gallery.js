import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Gallery extends Component {

  constructor(props) {
    super(props);
    
    this.state = {
      index: 0
    }

    this.handleLeftButton = this.handleLeftButton.bind(this);
    this.handleRightButton = this.handleRightButton.bind(this);
  }

  static propTypes = {
    data: PropTypes.array.isRequired
  }

  handleLeftButton() {
    let decrementIndex = this.state.index;
    decrementIndex--;
    const index = decrementIndex === -1 ? this.props.data.length - 1 : decrementIndex;
    this.setState({ index });
  }

  handleRightButton() {
    let incrementIndex = this.state.index;
    incrementIndex++;
    const index = incrementIndex === this.props.data.length ? 0 : incrementIndex;
    this.setState({ index });
  }

  render() {
    let { data } = this.props;

    if (!data.length) return (
      <div>
        <img src='https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif' alt='loading'/>
      </div>
    );

    let currentImage = data[this.state.index];
    let { title, description, url } = currentImage;

    return (
      <div className='Gallery'>
        <div className='button-container'>
          <button onClick={this.handleLeftButton}>&larr;</button>
          <button onClick={this.handleRightButton}>&rarr;</button>
        </div>
        <div className='Gallery-view'>
          <h2>{title}</h2>
          <p>{description}</p>
          <img src={url} alt={title} className='Gallery-image' />
        </div>
      </div>
    );
  }
}

export default Gallery;