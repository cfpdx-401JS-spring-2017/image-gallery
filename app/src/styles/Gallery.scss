.gallery-btn {
  display: inline-block;
  width: 8%;
  text-align: center;
}