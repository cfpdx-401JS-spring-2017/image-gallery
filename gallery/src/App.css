.App {
  text-align: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.ThumbNail {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  height: 30vh;
}

.List {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

button {
  height: 8vh;
  width: 12vw;
  background-color: #2F3E46;
  color:#CAD2C5;
  border: none;
  margin: 2vh 2vh;
  box-shadow: none;
}
