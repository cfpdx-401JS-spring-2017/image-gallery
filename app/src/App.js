import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from 'react-router-dom';

import Home from './components/Home';
import Albums from './components/albums/Albums';
import './styles/App.css';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <h2>CPDB: Cute Pomeranian Database</h2>
          <p>
            Welcome to the <code>Cute Pomeranian Database</code>! Check out our poms by clicking below:
          </p>
          <Switch>
            <Route exact path='/' component={Home}/>
            <Route path='/albums' component={Albums} />
            <Redirect to='/' />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
