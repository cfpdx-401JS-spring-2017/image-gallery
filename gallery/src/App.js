import React, { Component } from 'react';
import './App.css';
import AlbumsViewer from './AlbumsViewer';
import Albums from './Albums';

import {
  BrowserRouter as Router,
  Route, Switch, Redirect, Link
} from 'react-router-dom'

class App extends Component {
  constructor() {
    super();
    this.state = {
    }

  }

  render() {
  
    return (
      <div className="App">
        <Router>
            <div>
                <Switch>
                    {/*<Route exact path="/" component={App}/>*/}
                    <Route path="/albums/:id" component={AlbumsViewer}/>
                    <Route path="/albums" component={Albums}/>
                    <Redirect path="/albums" component={Albums}/>
                </Switch>
                <button>
                  <Link to="/albums">Go to Albums</Link>
                </button>
            </div>
        </Router>
      </div>  
    );
  }
}

export default App;
