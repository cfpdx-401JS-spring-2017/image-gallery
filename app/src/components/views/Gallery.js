import React, { Component } from 'react';
import PropTypes from 'prop-types';
import GalleryImage from './GalleryImage';
import '../../styles/Gallery.css';

export default class Gallery extends Component {

  static propTypes = {
    poms: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      description: PropTypes.string,
      url: PropTypes.string.isRequired
    }))
  };

  constructor(props) {
    super(props);

    this.state = {
      index: 0
    };
  }

  changePom(increment) {
    const poms = this.props.poms;
    const pom = this.props.poms[this.state.index];
    let index = this.state.index;

    index += increment;
    if (index === poms.length) {
      index = 0;
    } if (index === -1 || isNaN(index)) {
      index = poms.length - 1;
    }
    this.setState({ poms: pom });
    this.setState({ index });
  }

  render() {
    const { poms } = this.props;
    const { index } = this.state;
    return (
      <div>
        <button className="button gallery-btn" onClick={() => (
          this.changePom(-1)
        )}>
          Previous
        </button>
        <GalleryImage
          poms={poms[index]}
          onDelete={() => this.props.handleDelete(poms[index]._id, index)}
        />
        <button className="button gallery-btn" onClick={() => (
          this.changePom(+1)
        )}>
          Next
        </button>
      </div>
    );
  }
}