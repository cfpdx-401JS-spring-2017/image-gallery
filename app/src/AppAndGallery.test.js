import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Gallery from './Gallery';
import ThumbNail from './ThumbNail';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders a snapshot of main App component', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a snapshot of Gallery component', () => {
  const component = renderer.create(<Gallery />);
  const bobo = component.toJSON();
  expect(bobo).toMatchSnapshot();
});


