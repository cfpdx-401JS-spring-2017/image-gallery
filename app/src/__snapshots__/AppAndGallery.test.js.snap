// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of Gallery component 1`] = `
<div
  className="Gallery"
>
  <p
    className="mainButtons"
    onClick={[Function]}
  >
    Go To Gallery
  </p>
</div>
`;

exports[`renders a snapshot of main App component 1`] = `
<div
  className="App"
>
  <div>
    <button>
      <a
        className="Link"
        href="/albums"
        onClick={[Function]}
      >
        Go to Albums
      </a>
    </button>
  </div>
</div>
`;
